### ISSUE TITLE
Rename from {en,de}code_www_component to {en,de}code_www_form_component

### ISSUE THREAD

**Issue by ksss** [zzak/mruby-uri#11]
`URI.{en,de}code_www_component` is renamed to `{en,de}code_www_form_component` from CRuby v1.9.2 https://github.com/ruby/ruby/commit/f626a17d8c8a2c2c0ce2c2a5cc1f3b6b1fc643c3

I think, We also should use `{en,de}code_www_form_component` only.

What kind of milestone should we go forward?

1. Simply rename.
1. For a while, use alias.
1. Alias with warning message

I think, 1. is better.
2. is not good way, Because we can't tell the user of the change.
3. We need new dependeny to another mgem.

**Comment by zzak** [comment]
What dependency do we need for 3?

I think 1 is best, but we probably break some applications, since mrbgem doesn't have lock mechanism for dependencies.

**Comment by ksss** [comment]
Probably, `Kernel#warn` method is best.
But now, There is nothing mgem implemented this (probably).

We can use `$stderr.puts` with mruby-io instead.

**Comment by zzak** [comment]
I'm ok with `$stderr.puts` for now

**Comment by ksss** [comment]
As yet another proposal, There is way using C API `mrb_warn` like this.

```c
static mrb_value
mrb_uri_s_decode_www_component(mrb_state *mrb, mrb_value self)
{
  mrb_value str;
  mrb_warn(mrb, "Deprecated warning: ...");
  mrb_get_args(mrb, "o", &str);
  return mrb_funcall(mrb, self, "decode_www_form_component", 1, str);
}
...
mrb_define_class_method(mrb, mrb_module_get(mrb, "URI"), "decode_www_component", mrb_uri_s_decode_www_component, MRB_ARGS_REQ(1));
```

It doesn't need dependency.

**Comment by zzak** [comment]
LGTM

**Comment by ksss** [comment]
Resolved from ff16f5800a1af7aec31f2ebe0446f4c5ac1f27a3
